great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -2300,10 +2303,22 @@ def add_or_update_expectation_suite(
             )
 
         if existing_suite:
-            self.update_expectation_suite(existing_suite)
-            return self.get_expectation_suite(
-                expectation_suite_name=expectation_suite_name, ge_cloud_id=id
+            self.delete_expectation_suite(
+                expectation_suite_name=existing_suite.name,
+                ge_cloud_id=existing_suite.ge_cloud_id,
             )
+            if id is None:
+                id = existing_suite.ge_cloud_id
+            if expectations is None:
+                expectations = existing_suite.expectations
+            if evaluation_parameters is None:
+                evaluation_parameters = existing_suite.evaluation_parameters
+            if data_asset_type is None:
+                data_asset_type = existing_suite.data_asset_type
+            if execution_engine_type is None:
+                execution_engine_type = existing_suite.execution_engine_type
+            if meta is None:
+                meta = existing_suite.meta
```

Comment:
Prefer the runtime args for updating but use the existing values as backup. To simplify things, we always delete and write a new value to the store.